Yosemite/YosemiteTests/Stores/OrderStoreTests.swift

```diff
@@ -791,6 +791,42 @@ final class OrderStoreTests: XCTestCase {
         ]
         assertEqual(expectedKeys, receivedKeys)
     }
+
+    func test_create_order_does_not_upsert_autodrafts() throws {
+        // Given
+        let expectation = self.expectation(description: "Create auto-draft order")
+        let store = OrderStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+        network.simulateResponse(requestUrlSuffix: "orders", filename: "order-auto-draft-status")
+
+        // When
+        let action = OrderAction.createOrder(siteID: sampleSiteID, order: sampleOrder()) { result in
+            XCTAssertTrue(result.isSuccess)
+            XCTAssertEqual(self.viewStorage.countObjects(ofType: Storage.Order.self), 0)
+            expectation.fulfill()
+        }
+        store.onAction(action)
```

Comment:
I prefer to use the `waitFor` helpers. It is more clear since assert checks would be in the "Then" section and not in the `When` one.

Reply:
Good point, updated in [52ae155](https://github.com/woocommerce/woocommerce-ios/pull/6307/commits/52ae155529e9bf65455908c65b459ebd83e64276).